js/draw.mjs

```diff
@@ -79,25 +81,64 @@ function toggleDrawCanvas() {
 
 function clear() {
   ctx.clearRect(0, 0, drawCanvas.width, drawCanvas.height);
+  drawCanvas.style.display='block';
+  previewImage.style.display='none';
 }
 
 function downloadFile() {
-  const a = document.createElement('a');
-  a.style.display = 'none';
-  a.href = drawCanvas.toDataURL('image/png');
-  a.download = 'diagram.png';
-  document.body.appendChild(a);
-  a.click();
+  if(drawCanvas.style.display==='none'){
+    const a = document.createElement('a');
+    a.style.display = 'none';
+    a.href = previewImage.src;
+    a.download = 'diagram.png';
+    document.body.appendChild(a);
+    a.click(); 
+  }
+  else{
+      const a = document.createElement('a');
+      a.style.display = 'none';
+      a.href = drawCanvas.toDataURL('image/png');
+      a.download = 'diagram.png';
+      document.body.appendChild(a);
+      a.click();
+  }
 }
 
 function addToPaper() {
-  document.querySelector('#note').innerHTML = /* html */`
-    <img style="width: 100%;" src="${drawCanvas.toDataURL('image/png')}" />
-  ` + document.querySelector('#note').innerHTML;
+
+  if(drawCanvas.style.display==='none')
+  {
+        const src=previewImage.src;
+        document.querySelector('#note').innerHTML = /* html */`
+        <img style="width: 40%;" src="`+src+`" />
+        `+ document.querySelector('#note').innerHTML;
+  }
+  else{
+        
+        document.querySelector('#note').innerHTML = /* html */`
+        <img style="width: 100%;" src="${drawCanvas.toDataURL('image/png')}" />
+      ` + document.querySelector('#note').innerHTML;
+  }
 
   toggleDrawCanvas();
 }
 
+function addImageToPaper(){
+  imagePath.value="";
+  imagePath.click();
+  imagePath.addEventListener('change',function(){
+    const file=this.files[0];
+    if(file){
+      const reader=new FileReader();
+      previewImage.style.display='block';
+      drawCanvas.style.display='none';
+      reader.addEventListener('load',function(){
+        previewImage.setAttribute("src",this.result);
```

Comment:
Instead of having a new previewImage element and having to write all the other extra functions to handle its download and addition to paper, we can have the following code here
```js
const tempImage = new Image();
tempImage.src = this.result;
drawCanvas.getContext('2d').drawImage(tempImage, 0, 0);
```

With this, the image will be added to canvas and the addition to page and setting download source will be handled by the existing code of handling canvas.

So the changes from addToPaper function and downloadFile function will not be needed I believe

Reply:
yes, thanks for suggestion @saurabhdaware I have refactored the code in my latest commit.